src/scanner.c

```diff
@@ -1901,6 +1903,52 @@ static bool inner_scan(void *payload, TSLexer *lexer, const bool *valid_symbols)
             }
             break;
 
+        case ':':
+            if (valid_symbols[UNQUOTED_SYMBOL]) {
+                lex_advance(lexer);
+
+                int32_t lookahead = lexer->lookahead;
+
+                if (('A' <= lookahead && lookahead <= 'Z')
+                    || ('a' <= lookahead && lookahead <= 'z')
+                    || (lookahead == '_')
+                    || (0x00a0 <= lookahead && lookahead <= 0x10ffffff)) {
+
+                    // This is the start of a symbol
+                    lexer->result_symbol = UNQUOTED_SYMBOL;
+                    lex_advance(lexer);
+
+                    while (is_ident_part(lexer->lookahead)) {
+                        lex_advance(lexer);
+                    }
+
+                    switch (lexer->lookahead) {
+                        case '?':
+                            // Symbols like `:foo?` always include the trailing character
+                            lex_advance(lexer);
+                            return true;
+
+                        case '!':
+                        case '=':
+                            // Symbols like `:foo!` or `:bar=` include the trailing character,
+                            // only if the next char isn't also `=`
+                            lexer->mark_end(lexer);
+                            lex_advance(lexer);
+
+                            if (lexer->lookahead != '=') {
+                                lexer->mark_end(lexer);
+                            }
+                            return true;
+
+                        default:
+                            return true;
+                    }
+                } else {
```

Comment:
Nitpick: else isn't necessary here as it can just fallback on the default return false

Reply:
True. I wonder if returning explictly and early is still the better style?